### ISSUE TITLE
kubernetes autoscaler failed: ' No such file or directory: '/root/.kube/config''

### ISSUE THREAD

**Issue by drivenow** [ray-project/ray#6079]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04)
- **Ray installed from (source or binary)**:binary
- **Ray version**:0.7.6
- **Python version**:3.6
- **Exact command to reproduce**:

### Describe the problem

I use the autoscaler/kubernetes/examples-full.yaml to create an autoscaler ray cluster.  In preparation, I just change  image in examples-full.yaml to 'ray-project/examples', and add some operations like 'pip install kubernetes' in that dockerfile.  Then, I execute 'ray up examples-full.yaml', and excute 'ray submit example-full.yaml tt.py ' to run a simple ray example.

**But I find that the ray cluster doesn't scale, my code only run on the head node.** So I execute 'ray exec /app/anaconda/lib/python3.6/site-packages/ray/autoscaler/kubernetes/example-full.yaml 'tail -n 100 -f /tmp/ray/session_*/logs/monitor*', **I get the following error**: 

=============================================================
2019-11-04 03:19:03,291	ERROR autoscaler.py:762 -- StandardAutoscaler: kill_workers triggered
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/kubernetes/__init__.py", line 18, in _load_config
    kubernetes.config.load_incluster_config()
  File "/opt/conda/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 95, in load_incluster_config
    cert_filename=SERVICE_CERT_FILENAME).load_and_set()
  File "/opt/conda/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 47, in load_and_set
    self._load_config()
  File "/opt/conda/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 64, in _load_config
    raise ConfigException("Service token file does not exists.")
kubernetes.config.config_exception.ConfigException: Service token file does not exists.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/ray/monitor.py", line 361, in run
    self._run()
  File "/opt/conda/lib/python3.6/site-packages/ray/monitor.py", line 348, in _run
    self.autoscaler.update()
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/autoscaler.py", line 465, in update
    raise e
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/autoscaler.py", line 457, in update
    self._update()
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/autoscaler.py", line 477, in _update
    nodes = self.workers()
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/autoscaler.py", line 732, in workers
    tag_filters={TAG_RAY_NODE_TYPE: NODE_TYPE_WORKER})
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/kubernetes/node_provider.py", line 42, in non_terminated_nodes
    pod_list = core_api().list_namespaced_pod(
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/kubernetes/__init__.py", line 27, in core_api
    _load_config()
  File "/opt/conda/lib/python3.6/site-packages/ray/autoscaler/kubernetes/__init__.py", line 20, in _load_config
    kubernetes.config.load_kube_config()
  File "/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 546, in load_kube_config
    config_persister=config_persister)
  File "/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 503, in _get_kube_config_loader_for_yaml_file
    with open(filename) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/root/.kube/config'

=============================================================

So to enable autoscaler, should I install kubernetes in the docker image and which kubenetes api should I configure?

**Comment by edoakes** [comment]
What version of Kubernetes are you running on?

Inside the head node we use the [kubernetes python client](https://github.com/kubernetes-client/python). Here we are calling `kubernetes.config.load_kube_config()`, which tries to load the configuration from a standard location (the directory in the error above). Is it possible that your cluster has a non-default layout for the k8s config files? You could try poking around to find the `.kube` directory.

**Comment by drivenow** [comment]
Hi @ebrevdo , Kubernetes version is 1.9.0. And I installed kubernetes python client in my docker container (ray head node).But I did not install  kubernetes in that docker container.
Exactly，I executed ‘ray up examples-full.yaml’ in my kubernetes cluter， and then a docker container (ray head node) started on one of the nodes in my kubernetes cluter. I did not install Kubernetes in the docker container, I just installed 'kubernetes python client' in the docker container.
-----------------------------------------------------------------------------------
So what should I do to automaticly scale up more ray docker containers? It seems that my ray head container can not connect to my kubernetes cluter because I did not install Kubernetes in it.

**Comment by drivenow** [comment]
Hi @ebrevdo , Kubernetes version is 1.9.0. And I installed kubernetes python client in my docker container (ray head node).But I did not install kubernetes in that docker container.
Exactly，I executed ‘ray up examples-full.yaml’ in my kubernetes cluter， and then a docker container (ray head node) started on one of the nodes in my kubernetes cluter. I did not install Kubernetes in the docker container, I just installed 'kubernetes python client' in the docker container.

So what should I do to automaticly scale up more ray docker containers? It seems that my ray head container can not connect to my kubernetes cluter because I did not install Kubernetes in it.
